ZCS-2727 : API to get the list of Contact backups

chinmay15 opened the pull request:
Problem : Write api to get list of contact backups

Fix : New api written to get list of contact bacups

Testing done :
1. Manual testing of api
2. Unit test added to verify the same

Testing to be done by QA :
1. Manual testing of api
2. Automation for the same. No other api is disturbed with this, so no other testing required.

fatkudu commented on soap/src/java/com/zimbra/soap/mail/message/GetContactBackupListResponse.java:
```diff
@@ -0,0 +1,90 @@
+     * @param backup the backup to set
+     */
+    public void setBackup(List<String> backup) {
+        this.backup = backup;
```
Suggest the constructor `GetContactBackupListResponse(List<String> backup)` calls this.
That would resolve the current inconsistency between them in handling empty lists.

chinmay15 replied:
done

fatkudu commented on store/src/java-test/com/zimbra/cs/service/mail/GetContactBackupListTest.java:
```diff
@@ -0,0 +1,64 @@
+                + "<backup>file4.tgz</backup>"
+                + "</GetContactBackupListResponse>";
+
+        Assert.assertEquals(expectedResponse, response.toString());
```
Please use the variants of Assert methods that describe what is being checked so that when it fails, the report generated is immediately useful instead of requiring digging in the source to find the line of failure.

chinmay15 replied:
done

fatkudu commented on store/src/java-test/com/zimbra/cs/service/mail/GetContactBackupListTest.java:
```diff
@@ -0,0 +1,64 @@
+                + "</GetContactBackupListResponse>";
+
+        Assert.assertEquals(expectedResponse, response.toString());
+    }
```
Worth testing the JSON variant as well?

chinmay15 replied:
done

chinmay15 commented on store/src/java/com/zimbra/cs/service/mail/GetContactBackupList.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    // TODO : Update this method along with contact backup functionality
+    private List<String> getContactBackupList(Account account) {
```
This parameter will be used when the functionality will be implemented.

chinmay15 commented on store/src/java/com/zimbra/cs/service/mail/GetContactBackupList.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    // TODO : Update this method along with contact backup functionality
+    private List<String> getContactBackupList(Account account) {
```
This parameter will be used when the functionality will be implemented.

chinmay15 commented:
Merged into develop

commit ca0eb1f0e6c989c449c1401d742ca73cf1cbc7fd
Author: Chinmay Pathak <15.chinmay@gmail.com>
Date:   Mon Sep 11 17:54:44 2017 +0530

    ZCS-2727 : API to get the list of Contact backups

    Resolved review comments, added json support and unit test, updated soap.txt
